src/exponentials.h

```diff
@@ -0,0 +1,333 @@
+  *expF1 = *expF1*x + p2;
+  *expF1 = *expF1*x + p1;
+  *expF1 = *expF1*x + p0;
+}
+
+
+/**
+ * @brief Computes most of the F2 exponential term.
+ * @details This method computes (x-2+exp(-x)*(2+x))/x**2 using a 8th order
+ *          polynomial approximation. It is accurate to 1e-5 over [0, 5]. It was
+ *          generated by Colin Josey using Remez's algorithm.
+ * @param x the value at which to evaluate the function, usually tau
+ * @param expH a pointer to the exponential
+ */
+inline void expF2_poly(FP_PRECISION x, FP_PRECISION* expF2) {
+
+  const FP_PRECISION p1 = 1.666656766985825*1e-1;
+  const FP_PRECISION p2 = -8.331262914972137*1e-2;
+  const FP_PRECISION p3 =  2.492325839109710*1e-2;
+  const FP_PRECISION p4 = -5.440953156443790*1e-3;
+  const FP_PRECISION p5 =  9.034802244154581*1e-4;
+  const FP_PRECISION p6 = -1.091608727341864*1e-4;
+  const FP_PRECISION p7 =  8.411465095972204*1e-6;
+  const FP_PRECISION p8 = -3.029020287833148*1e-7;
+
+  *expF2 = p8*x + p7;
+  *expF2 = *expF2*x + p6;
+  *expF2 = *expF2*x + p5;
+  *expF2 = *expF2*x + p4;
+  *expF2 = *expF2*x + p3;
+  *expF2 = *expF2*x + p2;
+  *expF2 = *expF2*x + p1;
+  *expF2 = *expF2*x;
+}
+
+
+/**
+ * @brief Computes most of the H exponential term.
+ * @details This method computes (1-exp(-x)*(1+x))/x**2 using a 8th order
+ *          polynomial approximation. It is accurate to 1e-5 over [0, 5]. It was
+ *          generated by Colin Josey using Remez's algorithm.
+ * @param x the value at which to evaluate the function, usually tau
+ * @param expH a pointer to the exponential
+ */
+inline void expH_poly(FP_PRECISION x, FP_PRECISION* expH) {
+
+  const FP_PRECISION p0 = 0.5;
+  const FP_PRECISION p1 = -3.33307480097059 *1e-1;
+  const FP_PRECISION p2 =  1.248597190564637*1e-1;
+  const FP_PRECISION p3 = -3.305711771115656*1e-2;
+  const FP_PRECISION p4 =  6.667701492411682*1e-3;
+  const FP_PRECISION p5 = -1.028726890908856*1e-3;
+  const FP_PRECISION p6 =  1.145870114989106*1e-4;
+  const FP_PRECISION p7 = -8.079532805720403*1e-6;
+  const FP_PRECISION p8 =  2.657707693467421*1e-7;
+
+  *expH = p8*x + p7;
+  *expH = *expH*x + p6;
+  *expH = *expH*x + p5;
+  *expH = *expH*x + p4;
+  *expH = *expH*x + p3;
+  *expH = *expH*x + p2;
+  *expH = *expH*x + p1;
+  *expH = *expH*x + p0;
+}
+
+
+/**
+ * @brief Computes an exponential G term, used to reconstruct F1, F2 and H.
+ * @details This method computes 1/x-(1-exp(-x))/x**2 using a 5/6th order
+ *          rational approximation. It is accurate to 2e-7 over [0, 1e5].
+ *          However, accuracy when reconstructing F1, F2 and H is lower.
+ *          It was generated by Colin Josey using Remez's algorithm.
+ * @param x the value at which to evaluate the function, usually tau
+ * @param expH a pointer to the exponential
+ */
+inline void expG_fractional(FP_PRECISION x, FP_PRECISION* expG) {
+
+  /* Coefficients for numerator */
+  const FP_PRECISION p0 = 0.5;
+  const FP_PRECISION p1 = 1.76558112351595 * 1E-1;
```

Comment:
Upper case or lower case E, just keep the same.

Reply:
Ok keeping the same convention for all routines now